test/test-fs-copyfile.c

```diff
@@ -188,7 +188,7 @@ TEST_IMPL(fs_copyfile) {
   unlink(dst);
   r = uv_fs_copyfile(NULL, &req, fixture, dst, UV_FS_COPYFILE_FICLONE_FORCE,
                      NULL);
-  ASSERT(r == 0 || r == UV_ENOSYS || r == UV_ENOTSUP);
+  ASSERT(r <= 0 || r == UV_ENOSYS);
```

Comment:
UV_ENOSYS is < 0 so all this can just be `ASSERT(r <= 0)`

Reply:
Ah, ok, thanks. It might be more readable to keep at least UV_ENOSYS, though. Which makes me wonder if the ASSERT() has any effect at all here if it is done this way.

What about listing the `ioctl()` error codes separately? On the machines I have access to, there are currently 4 of them. Still manageable but it **may** vary with kernel version.

Reply:
> Which makes me wonder if the ASSERT() has any effect at all here if it is done this way.

It's at least a little bit useful in that we can verify an expected range of values (for example, no positive numbers).

> What about listing the ioctl() error codes separately?

If you go that route, [this Node.js test](https://github.com/nodejs/node/blob/88bff82624628df261a429386d81565023062e44/test/parallel/test-fs-copyfile.js#L71-L72) might help - although the changes in this PR might alter that list.

Reply:
> > Which makes me wonder if the ASSERT() has any effect at all here if it is done this way.
> 
> It's at least a little bit useful in that we can verify an expected range of values (for example, no positive numbers).

Agreed.

> > What about listing the ioctl() error codes separately?
> 
> If you go that route, [this Node.js test](https://github.com/nodejs/node/blob/88bff82624628df261a429386d81565023062e44/test/parallel/test-fs-copyfile.js#L71-L72) might help - although the changes in this PR might alter that list.

Thanks for the pointer!
It is actually a bit more complicated than that, the recent implementation apparently uses [`ioctl_file_clone()`](https://github.com/torvalds/linux/blob/3b1f00aceb7a67bf079a5a64aa5c6baf78a8f442/fs/ioctl.c#L222-L245) which returns `0`, `-EBADF`, `-EXDEV`, `-EINVAL`, or the error raised by [`vfs_copy_file_range()`](https://github.com/torvalds/linux/blob/3b1f00aceb7a67bf079a5a64aa5c6baf78a8f442/fs/read_write.c#L1625-L1687) which has a longer list and elevates the underlying return values itself.

I think I'd go with `<= 0` then.

I tried to find actual code, apart from tests and documentation, that uses that feature (`FORCE`).